odoo/addons/base/models/qweb.py

```diff
@@ -655,115 +330,214 @@ def _compile_node(self, el, options):
-                        self._append(ast.Call(
-                            func=ast.Name(id='escape', ctx=ast.Load()),
-                            args=[ast.Call(
-                                func=ast.Name(id='to_text', ctx=ast.Load()),
-                                args=[ast.Name(id='value', ctx=ast.Load())], keywords=[],
-                                starargs=None, kwargs=None
-                            )], keywords=[],
-                            starargs=None, kwargs=None
-                        )),
-                        self._append(ast.Str(u'"')),
-                    ],
-                    orelse=[]
-                )],
-                orelse=[]
-            )
-        ]
+    def _compile_options(self, el, varname, options, indent):
+        """
+        compile t-options and add to the dict the t-options-xxx values
+        """
+        t_options = el.attrib.pop('t-options', None)
+        directive = ''
+        if t_options:
+            directive = f't-options="{t_options}"'
+
+        code = []
+        dict_arg = []
+        for key in list(el.attrib):
+            if key.startswith('t-options-'):
+                value = el.attrib.pop(key)
+                directive += f' {key}="{value}"'
+                option_name = key[10:]
+                dict_arg.append(f'"{self._compile_str(option_name)}":{self._compile_expr(value)}')
+
+        if t_options and dict_arg:
+            code.append(self._indent(f"{varname} = {{**{self._compile_expr(t_options)}, {', '.join(dict_arg)}}}", indent))
+        elif dict_arg:
+            code.append(self._indent(f"{varname} = {{{', '.join(dict_arg)}}}", indent))
+        elif t_options:
+            code.append(self._indent(f"{varname} = {self._compile_expr(t_options)}", indent))
+
+        if code:
+            return self._compile_hook_before_directive(el, directive, options, indent) + \
+                code + \
+                self._compile_hook_after_directive(el, directive, options, indent)
+
+        return code
+
+    def _compile_format(self, expr):
+        """ Parses the provided format string and compiles it to a single
+        expression ast, uses string concatenation via "+".
+        """
+        text = '"'
+        base_idx = 0
+        for m in _FORMAT_REGEX.finditer(expr):
+            literal = expr[base_idx:m.start()]
+            if literal and isinstance(literal, str):
+                text += self._compile_str(literal)
+            text += f'" + to_text({self._compile_expr(m.group(1) or m.group(2))}) + "'
+            base_idx = m.end()
+        # string past last regex match
+        literal = expr[base_idx:]
+        if literal and isinstance(literal, str):
+            text += self._compile_str(literal)
+        text += '"'
+        return text
+
+    def _compile_expr(self, expr, raise_on_missing=False):
+        result = []
+
+        tokens = list(tokenize.tokenize(io.BytesIO(expr.strip().encode('utf-8')).readline))
```

Comment:
1. `print ([x for x in tokenize.generate_tokens(expr)` fail because str is not callable. We should not remove readline. When I test with `generate_tokens` (and readline) I had `TypeError: 'in <string>' requires string as left operand, not int`. The documentation I found does not give much information. I prefer to keep what works.
2. I'll add a try/except catching and raise a ValueError